### PR TITLE
[XPU] amp supports XPU bf16

### PR DESCRIPTION
**Pull request by dynamicheart**
### PR Category

Custom Device 

### PR Types

New features

### Description

amp supports XPU3 bf16

These following tests haven't been added for XPU:
- `test/amp/test_collect_operator_stats.py`
- `test/amp/test_compare_accuracy_api.py`
- `test/amp/test_tensor_checker.py`

### CODE REVIEW

File: python/paddle/amp/auto_cast.py [hunk 1]
```diff
@@ -761,12 +761,6 @@ def amp_decorate(
             return models
         else:
             return models, optimizers
-    # For xpu:
-    if tracer._expected_place.is_xpu_place() and (dtype == 'bfloat16'):
```
Comment (dynamicheart) [inline python/paddle/amp/auto_cast.py:764]: 是否需要增加一个`_is_xpu_bfloat16_supported`来区分处理`xpu1`、`xpu2`、`xpu3`的情况？

**Review by houj04 — APPROVED** [review]

**Review by qili93 — APPROVED** [review]
LGTM

**Review by runzhech — COMMENTED** [review]
lgtm

**Review by runzhech — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Comment by dynamicheart** [comment]
PR-CI-Coverage测试由于修改的是XPU，因此没有执行无法覆盖到，需要进行豁免

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**